docs/pages/technical-specs/expo-updates-0.md

```diff
@@ -107,18 +107,20 @@ type Asset = {
   hash: string;
   key: string;
   contentType: string;
+  fileExtenstion?: string;
   url: string;
 }
 ```
   * `id`: The ID MUST uniquely specify the manifest.
   * `createdAt`: The date and time at which the update was created is essential as the client library selects the most recent update (subject to any constraints supplied by the `expo-manifest-filters` header). The datetime should be formatted according to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601).
   * `runtimeVersion`: Can be any string defined by the developer. It stipulates what native code setup is required to run the associated update.
-  * `launchAsset`: A special asset that is the entry point of the application code.
+  * `launchAsset`: A special asset that is the entry point of the application code. The `fileExtension` field will be ignored for this asset and SHOULD be omitted.
   * `assets`: An array of assets used by the update bundle, such as JavaScript, pictures, and fonts. All assets (including the `launchAsset`) should be downloaded to disk before executing the the update, and a mapping of asset `key`s to locations on disk should be provided to application code.
   * Properties of each asset object:
     * `hash`: SHA-256 hash of the file to guarantee integrity.
     * `key`: Key used to reference this asset from the update's application code. This key, for example, may be generated by a separate build step that processes the application code, such as a bundler.
     * `contentType`: The MIME type of the file as defined by [RFC 2045](https://tools.ietf.org/html/rfc2045). e.g. `application/javascript`, `image/jpeg`.
+    * `fileExtension`: The suggested extension to use when a file is saved on a client. Some platforms, such as iOS, require certain file types to be saved with an extension. The extension MUST be prefixed with a `.`. e.g. `.jpeg`. In some cases, such as the launchAsset, this field will be ignored in favor of a locally determined extension. If the field is omitted and there is no locally stipulated extensiion, the asset will be saved without an extension, e.g. `./filename` with no `.` at the end.
```

Comment:
```suggestion
    * `fileExtension`: The suggested extension to use when a file is saved on a client. Some platforms, such as iOS, require certain file types to be saved with an extension. The extension MUST be prefixed with a `.`. e.g. `.jpeg`. In some cases, such as the launchAsset, this field will be ignored in favor of a locally determined extension. If the field is omitted and there is no locally stipulated extension, the asset will be saved without an extension, e.g. `./filename` with no `.` at the end.
```

Also, what happens if the extension doesn't start with "."? Does the client ignore the file extension? Does it crash (seems bad)? Does it just concatenate it anyway? I think we should make the decision based on what the best outcome at runtime is.

Reply:
I don't see any valid situation in which we would want to concatenate a name and a "dotless" file extension without adding a dot.

Seems like we should follow the robustness principle: "be conservative in what you send, be liberal in what you accept"

Our server only sends dotted file extensions, but our client will handle both.

Perhaps worth adding a sentence like: 'For robustness, a client SHOULD prepend a `.` to a file extension missing a `.` prefix.'

Reply:
We should also document and implement the behavior of what happens when there isn't a dot

Reply:
Added descriptions for server and client.

* server should 400 (https://github.com/expo/universe/pull/7970)
* client should attempt to recover. (in progress)